### ISSUE TITLE
BO - Store page -the state field is displayed even though the country has no state ! 

### ISSUE THREAD

**Issue by djoelleuch** [PrestaShop/PrestaShop#32316]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

The state box is displayed even if the country have no State .

### Expected behavior

- The State box should be displayed only when the country already chosen has states !
- When the country chosen has states , the state box is well displayed 
- The store infiormation resigtration should only be skipped when all required fields have been completed 

### Steps to reproduce

1. Go to internationnal >locations >States >see there is only "United States " with the status 'enabled ' 
2. Go to Shop parameters >Contact >Stores >Add new store

fil the informations :  
Adress :Paris 
City : Paris 
Country : France 

>> SEE the state  box is displayed even though this country 'France' have no State 

with DEVELOP --> NOK    

[dev.webm](https://user-images.githubusercontent.com/115152460/234211259-c9578d50-245a-49b5-83e4-bd63c88d315e.webm)

with 8.0.3 --> OK 

[803.webm](https://user-images.githubusercontent.com/115152460/234211371-ff7e5c95-fa8d-4284-8497-790b1b5c84d4.webm)

### PrestaShop version(s) where the bug happened

develop

### PHP version(s) where the bug happened

7.4

**Comment by florine2623** [comment]
Hello @djoelleuch ,

I reproduce the issue with PrestaShop version `develop`, I'll add this to the backlog so it can be fixed.
On PS `8.0.x` and `8.1.x`, it was working well.

Please be aware that some issues might take a very long time to be resolved. If this one is important to you and you cannot wait for it to be fixed on the project’s own time, we strongly suggest you consider [contacting a professional](https://www.prestashop-project.org/support/) to help you. 

If you fix the issue on your end, please [contribute it back to the project](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/). Remember that the more people contribute, the better PrestaShop becomes for everyone.

Thank you

**Comment by tesshsu** [comment]
Hi,
Just check quickly, since this is call ajax from function ajaxStates, the require AdminStates controller is not exist in develope branch: https://github.com/PrestaShop/PrestaShop/tree/develop/controllers/admin

And if check version 8.0.3 ( which has no load states issue ) there's AdminSatesController : https://github.com/PrestaShop/PrestaShop/blob/8.0.3/controllers/admin/AdminStatesController.php

I think there's some commit removed controller/admin/AdminStatesController by accident
If this AdminSatesController add back to develop branch then it should be work:

![1](https://github.com/PrestaShop/PrestaShop/assets/3927152/6053756e-1e19-4809-9b0d-5fb190059dfb)

**Comment by matks** [comment]
> I think there's some commit removed controller/admin/AdminStatesController by accident

It was not an accident ;) the page has been migrated to Symfony.

**Comment by tesshsu** [comment]
Thanks for the feedback, just wondering if the new action path could be replaced by this route ? 
```
admin_country_states:
  path: /country-states
  methods: [ GET ]
  defaults:
    _controller: PrestaShopBundle\Controller\Admin\Improve\International\StateController::getStatesAction
    _legacy_controller: AdminStates
```

https://github.com/PrestaShop/PrestaShop/blob/8.0.3/src/PrestaShopBundle/Resources/config/routing/admin/improve/international/state.yml
tks

**Comment by jolelievre** [comment]
Fixed by #32554
